llama_stack/distribution/routing_tables/common.py

```diff
@@ -34,7 +34,8 @@ async def register_object_with_provider(obj: RoutableObject, p: Any) -> Routable
     assert obj.provider_id != "remote", "Remote provider should not be registered"
 
     if api == Api.inference:
-        return await p.register_model(obj)
+        return await p.register_model(model=obj, provider_id=obj.provider_id)
```

Comment:
this should not be needed?

Reply:
Which part? Without this, the provider doesn't get the provider_id.
Or are you saying we can do this differently?